### PR TITLE
Use blockly and closure library as git submodules

### PR DESCRIPTION
**Pull request by philippejadin**
This pull request is imho a good start to enhance blocklyduino from latest google blockly. It is based on Sebastien work on BlocklyDuino reboot, but instead use separates directories for everything and keep blockly untouched.

The provided /blocklyduino/index.html works without errors afaict

### CODE REVIEW

**Comment by SebCanet** [comment]
Great work! I think modularity is needed to enhance project based upon Blockly and to update easily.
But as I posted (https://github.com/BlocklyDuino/BlocklyDuinoReboot/issues) there are still problems with my generator update...

**Comment by philippejadin** [comment]
let's work some more on it then. I also have some fears with git submodules, it seems they might bite us later... trying with npm dependencies system

File: README.md [hunk 1]
```diff
@@ -58,6 +58,15 @@ OR (if running `ino_web_server.py`):
 1. Open browser to BlocklyDuino, drag and drop blocks to make an Arduino program.
 2. Select the 'Arduino' tab and press the 'Upload' button. (press the 'Reset' button to upload an empty program)
 
+### Developpers
+To work on BlocklyDuino, you need google blockly, google closure and python
+
+- Clone the repository with submodules : `git clone --recurse-submodules https://github.com/philippejadin/BlocklyDuino`
```
Comment (gasolin) [inline README.md]: could be point to https://github.com/BlocklyDuino/BlocklyDuino

**Review by gasolin — CHANGES_REQUESTED** [review]
Looks great! will test more this week!

@here I encourage everyone interest in BlocklyDuino test this PR

File: README.md (hunk 1 above)
Comment (philippejadin) [reply README.md]: Yes ! I wanted to be sure people clone the correct repository, chicken and egg problem. But I'll correct it already.

Note that for this to work, people also need to checkout the v2 branch. On my fork, I seted up v2 as the main branch. 
```suggestion
- Clone the repository with submodules : `git clone --recurse-submodules https://github.com/BlocklyDuino/BlocklyDuino`
- Checkout the v2 branch : `git checkout v2`
```

**Comment by philippejadin** [comment]
There is still a lot to do of course. Like for example deciding how we handle translations. I'd like something that merge our own translations with those from blockly.

Would you find acceptable to define a `trans(localization_key)` function that would just take an localization_key and return the correct translated string? It would mean block must be defined in js (and not json) but it would be easy to code and future proof.

File: README.md [hunk 2]
```diff
@@ -58,6 +58,16 @@ OR (if running `ino_web_server.py`):
 1. Open browser to BlocklyDuino, drag and drop blocks to make an Arduino program.
 2. Select the 'Arduino' tab and press the 'Upload' button. (press the 'Reset' button to upload an empty program)
 
+### Developpers
+To work on BlocklyDuino, you need google blockly, google closure and python
+
+- Clone the repository with submodules : `git clone --recurse-submodules https://github.com/BlocklyDuino/BlocklyDuino`
+- Checkout the v2 branch : `git checkout v2`
```
Comment (gasolin) [inline README.md:65]: We can do clone in 3 steps as well
1. Clone the repository `git clone https://github.com/BlocklyDuino/BlocklyDuino`
2. Checkout the v2 branch : `git checkout v2`
3. Update submodules `git submodule update --init`

**Comment by gasolin** [comment]
> There is still a lot to do of course. Like for example deciding how we handle translations. I'd like something that merge our own translations with those from blockly.
> 
> Would you find acceptable to define a `trans(localization_key)` function that would just take an localization_key and return the correct translated string? It would mean block must be defined in js (and not json) but it would be easy to code and future proof.

@philippejadin could you elaborate more about this, or provide a mocked input/output for example?

Blockly now have their official translating guide 
- https://developers.google.com/blockly/guides/modify/web/translating
- https://translatewiki.net/wiki/Translating:Blockly

At this moment the priority is making v2 BlocklyDuino core function work with Blockly repo,
Then, we can make a script to append our translation file onto blockly's current msg file,  maintain our translation file (probably `en` only) by hand.
Then, we can adopt the same translation site used by `Blockly`, 
and create a separate `BlocklyDuino` msg list.
When we have some new updated translation, we can export it to our translation file.

**Comment by gasolin** [comment]
I merged this PR to v2 branch https://github.com/BlocklyDuino/BlocklyDuino/tree/v2
Welcome to send PR to move BlocklyDuino forward

**Comment by philippejadin** [comment]
Great thanks!

Will open a new issue to discuss localization

**Pull request merged**